bisum/tests/test_bisum.py

```diff
@@ -6,98 +6,70 @@
-#c = bisum( einsumstr, A.to_sparse(), B.to_sparse() )
-
-#print(C.shape, c.shape)
-#print( torch.allclose(C, c.to_dense()) )
-
-einsumstr = "aeacec,cdd -> aed" ## SPARSE-SPARSE, NO post-inter-externals, NO post-transpose
-A = torch.rand(10,10,10,10,10,10, device=device)
-B = torch.rand(10,10,10, device=device)
-print( torch.allclose( bisum( einsumstr, A.to_sparse(), B.to_sparse()).to_dense(), torch.einsum( einsumstr , A, B )) )
-
-einsumstr = "daa,aed -> a" ## SPARSE-SPARSE, YES post-inter-externals, NO post-transpose
-A = torch.rand(10,10,10, device=device)
-B = torch.rand(10,10,10, device=device)
-print( torch.allclose( bisum( einsumstr, A.to_sparse(), B.to_sparse()).to_dense(), torch.einsum( einsumstr , A, B )) )
-
-### acc -> ac
-### cdd -> cd  ----> accd ----> acd
-
-einsumstr = "daa,aed -> ea" ## SPARSE-SPARSE, YES post-inter-externals, YES post-transpose
-A = torch.rand(10,10,10, device=device)
-B = torch.rand(10,10,10, device=device)
-print( torch.allclose( bisum( einsumstr, A.to_sparse(), B.to_sparse()).to_dense(), torch.einsum( einsumstr , A, B )) )
-
-### acc -> ac
-### cdd -> cd  ----> accd ----> acd
-
-##### if slicing returns an empty array avoid lexsort!!!!!!!!!!!!!!!! AND "dad,mom -> dm"
-einsumstr = "aaa,wop -> a" ## SPARSE-SPARSE, YES post-inter-externals, YES post-transpose
-A = torch.rand(13,13,13, device=device)
-B = torch.rand(10,10,10, device=device)
-print( torch.allclose( bisum( einsumstr, A.to_sparse(), B.to_sparse()).to_dense(), torch.einsum( einsumstr , A, B )) )
-
-
-##### if slicing returns an empty array avoid lexsort!!!!!!!!!!!!!!!! AND "dad,mom -> dm"
-einsumstr = "aaa,wop -> a" ## SPARSE-DENSE, YES post-inter-externals, YES post-transpose
-A = torch.rand(10,10,10, device=device)
-B = torch.rand(10,10,10, device=device)
-print( torch.allclose( bisum( einsumstr, A.to_sparse(), B).to_dense(), torch.einsum( einsumstr , A, B )) )
-
-##### if slicing returns an empty array avoid lexsort!!!!!!!!!!!!!!!! AND "dad,mom -> dm"
-einsumstr = "qaq,wow -> " ## SPARSE-DENSE, YES post-inter-externals, YES post-transpose
-A = torch.rand(10,10,10, device=device)
-B = torch.rand(10,10,10, device=device)
-print( torch.allclose( bisum( einsumstr, A.to_sparse(), B).to_dense(), torch.einsum( einsumstr , A, B )) )
\ No newline at end of file
+
+
+# set up tensors for all tests (do NOT modify the tensors), treat them
+# as readonly; if tensors should be changed then change scope to
+# "function"
+@pytest.fixture(scope="module")
+def A_big():
+    return torch.rand(10,10,10,10,10,10, device=device)
+
+@pytest.fixture(scope="module")
+def A():
+    return torch.rand(10,10,10, device=device)
+
+@pytest.fixture(scope="module")
+def B():
+    return torch.rand(10,10,10, device=device)
+
+@pytest.fixture(scope="module")
+def C():
+    return torch.rand(13,13,13, device=device)
+
+
+
+@pytest.mark.parametrize("einsumstr", [
+    "aeacec,cdd -> aed",  #SPARSE-SPARSE, NO post-inter-externals, NO post-transpose
```

Comment:
Note that the tests can be parametrized so you can just add another `einsumstr` and it will run as its own tests with no additional coding necessary.